check_nightly_success/check-nightly-success/check.py

```diff
@@ -0,0 +1,74 @@
+import requests
+import itertools
+from datetime import datetime
+import sys
+import argparse
+import os
+
+# Constants
+GITHUB_TOKEN = os.environ["RAPIDS_GH_TOKEN"]
+GOOD_STATUSES = ("success",) # noqa
+
+
+def main(
+    repo: str,
+    repo_owner: str,
+    workflow_id: str,
+    max_days_without_success: int,
+):
+    headers = {"Authorization": f"token {GITHUB_TOKEN}"}
+    url = f"https://api.github.com/repos/{repo_owner}/{repo}/actions/workflows/{workflow_id}/runs"
+    response = requests.get(url, headers=headers)
+    response.raise_for_status()
```

Comment:
does `requests` have a retry mechanism for transient network failures?

Reply:
Nothing direct that I am aware of, so I added a manual retry loop.